particle-accelerator/src/renderer/utilities/ActivationAttempt.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * copy: performs a deep copy of a string to prevent reference storing
+ * @param input: the string to be copied
+ */
+function copy(input) {
+  return JSON.parse(JSON.stringify(input))
+}
+
+/**
+ * ActivationAttempt: Class for creating ActivationAttempt objects which stores
+ * the data of a device activation.
+ */
+export default class ActivationAttempt {
+  serialNumber
+
+  deviceName
+
+  productID
+
+  deviceID
+
+  iccid
+
+  country
+
+  SIMActivationStatus
+
+  namingStatus
+
+  totalStatus
+
+  timeStamp
+
+  constructor(serialNumber, deviceName, productID, deviceID, iccid, country, SIMActivationStatus, namingStatus, totalStatus) {
+    this.serialNumber = copy(serialNumber)
+    this.deviceName = copy(deviceName)
+    this.deviceID = copy(deviceID)
+    this.productID = copy(productID)
+    this.iccid = copy(iccid)
+    this.country = copy(country)
+    this.SIMActivationStatus = copy(SIMActivationStatus)
+    this.namingStatus = copy(namingStatus)
+    this.totalStatus = copy(totalStatus)
+    this.timeStamp = new Date()
+  }
```

Comment:
Yeah, this was kind of poor code looking back on it. I had a bug that was was being caused by passing the wrong field, so it made it look like the reference was being copied into the field instead of the value, so I tried to use this workaround to decouple any sort of references. Looking back at it now, I've removed the function as it is not necessary. Need to turn off my Java/reference-oriented brain.